classes/Mission.php

```diff
@@ -129,7 +129,7 @@ public function nextStage($stage_id) {
         if($this->current_stage['action_type'] == 'travel' || $this->current_stage['action_type'] == 'search') {
             for($i = 0; $i < 3; $i++) {
                 $location = $this->rollLocation($this->player->village_location);
-                if(!isset($villages[$location]) || $location == $this->player->village_location) {
+                if(!isset($villages[$location->fetchString()]) || $location->equals(TravelCoords::fromDbString($this->player->village_location))) {
```

Comment:
If we made `player->village_location` also an instance of `TravelCoords` we could save a lot of effort on all the places that check it, and further enhance our type safety